### DISCUSSION TITLE
change repo address

### DISCUSSION THREAD

**Discussion by n0p0n3** [AppOutlet/AppOutlet#226] [Q&A] [answered]
is it possible to change the repo address, like if you wanted to create a custom repo for appimages created a specific way? For example, if the system is setup with readonly root filesystem like a docker image with app-outlet on it and used it as the main "app-store", but you want to curate everything that goes on there and will build the appimages yourself. And how hard would it be to build app-outlet without snap or flatpack support, just appimage support only?

**Comment by MessiasLima** [comment, marked as answer]
You can change the place on the application fetch the app images by changing the url on [AppImageHub repository](https://www.github.com/app-outlet/app-outlet/tree/main/core%2Frepository%2Fappimagehub%2FAppImageHubRepository.js)

You just have to pay attention to the mappings on `core/service/synchronization/synchronizer/AppImageHubSynchronizer.js`

Your app data -> app outlet application data structure -> database


_____

To disable the other package types, you just need to remove the corresponding synchronizer call on `core/service/synchronization/SynchornizationService.js` on `synchronize()` method .

**Reply by n0p0n3** [reply]
Hey thanks! I've edited syncronizationservice.js function to only load appimage and can see the url in appimagehubrepository.js to edit to mine when I'm ready, but should I also edit the appimagehubsyncronizer.js? I also dont quite understand the mappings. Are they subdirs I need to have in my url when I create it? And is there somewhere that can explain setting it up like its done for y'all? I'm not well versed in github (mostly use my private one to store ideas and stuff for when my pc dies) I have no idea really how to set it up for this. I've tried looking up how to set up a repo (in my mind like how debian repos store apps), but i mostly get stuff about setting up react and have no idea what that is or if what I'm reading even has anything to do with what I want. Thanks again, been very helpfull
